dts/bindings/pinctrl/nxp,imx-iomuxc-scu.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Use this compatible for i.MX boards on which the
+  IOMUXC is managed by the SCU.
+
+compatible: "nxp,imx-iomuxc-scu"
+
+include: base.yaml
+
+child-binding:
+  description: SCFW-based IOMUXC pin mux.
+  properties:
+    pinmux:
+      required: true
+      type: array
+      description: |
+        This is an array of values defining the pin mux selection
+        with the following format:
+
+          <pad, mux, sw_config, lp_config, drive_strength, pull_selection>
+
+        pad: Which pad to configure.
+        mux: Select which signal to route.
+        sw_config: I/O configuration.
+        lp_config: Low power configuration.
+        drive_strength: Pad drive strength. (optional)
+        pull_selection: Pull configuration. (optional)
```

Comment:
why are some properties being encoded in non-standard way? ie pulls, strength, etc.

Reply:
The standard way of encoding these properties would work really well with drive strength and pull selection but not so much with the I/O configuration (`sw_config`) and the low power configuration (`lp_config`). For instance, to support all I/O configuration cases you'd have to use a mix of the `drive-open-drain`, `input-enable`, and `output-enable` properties (e.g: no property => NORMAL mode, `drive-open-drain` => OD mode with input or output buffer enabled, `drive-open-drain` + `input-enable` => OD mode with input buffer enabled (output buffer may be enabled or not) etc...). For me  this seems much messier to do rather than just using the macros provided in the pinctrl header.
Also, in the case of `lp_config`, there seems to be only one configuration supported? (i.e: `bias-bus-hold`)

I guess the "better" alternative would be to just encode all of the configurations in a single field in the pinmux property (as Linux seems to do) but if this is not a deal-breaker it can be addressed later on I guess.

My knowledge of the pad configuration is rather modest so please let me know if I've got something wrong.

Reply:
This is a requirement for new bindings, see https://docs.zephyrproject.org/latest/hardware/pinctrl/index.html#pin-control-drivers

Exceptions include implementations following an alternative convention followed elsewhere (eg Linux), provided there's a 1:1 matching (no Zephyr tweaks).

Reply:
Thanks for letting me know. Should be better now.

Reply:
Looking at the RM, it seems that the underlying IP behind the SCU is the IOMUXD- couldn't we declare custom properties where needed for certain pin configuration options? This is what was done for the IOMUXC peripheral: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/pinctrl/nxp%2Cmcux-rt-pinctrl.yaml. Please let me know if I'm missing something here, I'm not very familiar with this SOC or the SCU.

Reply:
I think this would also work, thanks for the suggestion! Currently, I've dropped the feature of allowing pad configuration as it was a bit of a headache and we weren't really using non-default configurations. This is more of a personal taste I guess but if possible I'd rather encode all of the configurations in a 32-bit field passed through the `pinmux` property instead of having to specify all of the standard properties (and custom ones) just like we do in Linux (seems more simple this way I guess?) (but this can change once I see the actual implementation). Anyways, at the moment this feature is more of a `nice to have` feature instead of a `must have` one so we'll deal with this later if everyone finds this acceptable.